src/hardware/kingst-la2016/protocol.c

```diff
@@ -723,9 +723,11 @@ SR_PRIV int la2016_start_retrieval(const struct sr_dev_inst *sdi, libusb_transfe
 	}
 
 	to_read = devc->n_bytes_to_read;
-	if (to_read > LA2016_BULK_MAX)
-		to_read = LA2016_BULK_MAX;
-
+	/* choose a buffer size for all of the usb transfers */
+	if (to_read >= LA2016_USB_BUFSZ)
+		to_read = LA2016_USB_BUFSZ; /* multiple transfers */
+	else /* one transfer, make buffer size some multiple of LA2016_EP6_PKTSZ */
+		to_read = (to_read + (LA2016_EP6_PKTSZ-1)) & ~(LA2016_EP6_PKTSZ-1);
```

Comment:
It probably makes sense to to turn the new code into an inline function? So we make sure that both invocations are always consistent.

Reply:
Your right, I should have done it that way. There are a few things that can be improved in the LA2016 driver, but at least it is functional for now. I would like to get the open mcu firmware I have for LA2016 accepted into sigrok, then look at tidy ups.